write-fonts/src/tables/glyf.rs

```diff
@@ -222,6 +222,57 @@ impl FontWrite for CoordDelta {
     }
 }
 
+/// A little helper for writing flags that may have a 'repeat' byte
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+struct RepeatableFlag {
+    flag: SimpleGlyphFlags,
+    repeat: u8,
+}
+
+impl FontWrite for RepeatableFlag {
+    fn write_into(&self, writer: &mut crate::TableWriter) {
+        debug_assert_eq!(
+            self.flag.contains(SimpleGlyphFlags::REPEAT_FLAG),
+            self.repeat > 0
+        );
+
+        self.flag.bits().write_into(writer);
+        if self.flag.contains(SimpleGlyphFlags::REPEAT_FLAG) {
+            self.repeat.write_into(writer);
+        }
+    }
+}
+
+impl RepeatableFlag {
+    /// given an iterator over raw flags, return an iterator over flags + repeat values
+    // writing this as an iterator instead of just returning a vec is very marginal
+    // gains, but I'm just in the habit at this point
+    fn iter_from_flags(
+        flags: impl IntoIterator<Item = SimpleGlyphFlags>,
+    ) -> impl Iterator<Item = RepeatableFlag> {
+        let mut iter = flags.into_iter();
+        let mut prev = None;
+
+        std::iter::from_fn(move || loop {
+            match (iter.next(), prev.take()) {
+                (None, prev) => return prev,
+                (Some(flag), None) => prev = Some(RepeatableFlag { flag, repeat: 0 }),
+                (Some(flag), Some(mut last)) => {
```

Comment:
The inherent logic for this is pretty nasty but this solution captures it nicely!

Reply:
Ew. Also +1.